### PR TITLE
feat: allow customizing the bench runner

### PR DESCRIPTION
**Pull request by katcipis**
I removed/skipped tests involving the use of "latest" behavior, testing with latest ended up generating tests that will be broken forever =/ (latest version changed and now the test fails since the wanted version doesn't match...doesn't seem like this was a good idea).

Ill add the tests back at some point...hopefully with better ideas/implementation (I was never super happy with them..and the latest usage was quite broken =/). And with a parametrized runner we can also test for corner cases more easily (like the empty bench test).

With support for custom commands it will be a nicer end to end/integration test to test the tool using this to generate deterministic results, we could test the integration of module downloading + parsing and the testing/check logic but not depending on any benchmark actually being run..since they exhibit a lot of variance...specially on CI/MacOS.

Still wondering what to do about the odd Windows 2019 error...probably just remove Windows 2019 from CI :laughing: (works on newer Windows).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-12, macos-11, ubuntu-20.04, windows-2019, windows-2022]
-        go: ["1.16", "1.17", "1.18", "1.19"]
+        os: [macos-12, macos-13, ubuntu-22.04, windows-2019, windows-2022]
+        go: ["1.16", "1.17", "1.18", "1.19", "1.20", "1.21"]
```
Comment (katcipis) [inline .github/workflows/test.yml:14]: Maybe we could remove some Go releases here, even though it works on older Go versions (make CI faster).

File: README.md [hunk 2]
```diff
@@ -39,26 +39,29 @@ have information on memory allocations.
 Comparing performance between two versions of a Go module
 and just showing results on output (no check performed):
 
-```
-benchcheck cool.go.module v0.0.1 v0.0.2
+```sh
```
Comment (katcipis) [inline README.md:42]: Docs were also wrong...sorry for that :-)

File: cmd/benchcheck/benchcheck.go [hunk 3]
```diff
@@ -66,7 +66,20 @@ func main() {
 		log.Fatal("-new is obligatory")
 	}
 
-	results, err := benchcheck.StatModule(*mod, *oldRev, *newRev)
+	runBench := benchcheck.DefaultRunBench
+	var customizedBenchCmd []string
+
+	for i, v := range os.Args {
+		if v == "--" {
+			customizedBenchCmd = os.Args[i+1:]
```
Comment (i4ki) [inline cmd/benchcheck/benchcheck.go:74]:
```suggestion
			customizedBenchCmd = os.Args[i+1:]
			break
```

because if the user-supplied command also contains a `--` it's going to use the last one, which seems wrong. Example:
```
benchcheck -mod A -old B -new C -- wrapper -flag -- go test ...
```

File: cmd/benchcheck/benchcheck.go [hunk 4]
```diff
@@ -66,7 +66,20 @@ func main() {
 		log.Fatal("-new is obligatory")
 	}
 
-	results, err := benchcheck.StatModule(*mod, *oldRev, *newRev)
+	runBench := benchcheck.DefaultRunBench
+	var customizedBenchCmd []string
+
+	for i, v := range os.Args {
```
Comment (i4ki) [inline cmd/benchcheck/benchcheck.go:72]: you can use [flag.Args()](https://pkg.go.dev/flag#Args) to only iterate in the non-flag remaining args.

File: .github/workflows/test.yml (hunk 1 above)
Comment (i4ki) [reply .github/workflows/test.yml:14]: yep, makes sense

**Review by i4ki — APPROVED** [review]

**Comment by i4ki** [comment]
Sorry for the very late feedback here :(
